src/overlays/actors/ovl_Bg_Breakwall/z_bg_breakwall.c

```diff
@@ -53,21 +66,263 @@ static ColliderQuadInit D_80870760 = {
     },
     { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Breakwall/BgBreakwall_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Breakwall/BgBreakwall_Init.s")
+static BombableWallInfo sBombableWallInfo[] = {
+    { 0x06000118, 0x06000040, 0 },
+    { 0x06000118, 0x06000040, 0 },
+    { 0x060264A8, 0x06025BD0, 1 },
+    { 0x06025B64, NULL, -1 },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+};
+
+void BgBreakwall_SetupAction(BgBreakwall* this, BgBreakwallActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgBreakwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgBreakwall* this = THIS;
+    s32 pad;
+    s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+    this->bombableWallDList = sBombableWallInfo[wallType].dList;
+    this->colType = sBombableWallInfo[wallType].colType;
+
+    if (this->colType == 1) {
+        this->dyna.actor.world.rot.x = 0x4000;
+    }
+
+    if (this->bombableWallDList != NULL) {
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F)) {
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+
+        ActorShape_Init(&this->dyna.actor.shape, 0.0f, NULL, 0.0f);
+        Collider_InitQuad(globalCtx, &this->collider);
+        Collider_SetQuad(globalCtx, &this->collider, &this->dyna.actor, &sQuadInit);
+    } else {
+        this->dyna.actor.world.pos.y -= 40.0f;
+    }
+
+    this->bankIndex = (wallType >= BWALL_KD_FLOOR) ? Object_GetIndex(&globalCtx->objectCtx, OBJECT_KINGDODONGO)
+                                                   : Object_GetIndex(&globalCtx->objectCtx, OBJECT_BWALL);
+
+    if (this->bankIndex < 0) {
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        BgBreakwall_SetupAction(this, BgBreakwall_WaitForObject);
+    }
+}
+
+void BgBreakwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgBreakwall* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+/**
+ * Spawns fragments using ACTOR_EN_A_OBJ whenever the wall or floor is exploded.
+ * Although it is not ever checked, it will also return the instance of the actor which it spawns.
```

Comment:
Imo, the part about the return value not being used is unnecessary so I would leave it out